scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java

```diff
@@ -1282,4 +1282,10 @@ public void setScheduledTimeForRemoval(long scheduledTimeForRemoval) {
         this.jobInfo.setScheduledTimeForRemoval(scheduledTimeForRemoval);
     }
 
+    @Override
+    public void setVariables(Map<String, JobVariable> variables) {
+        super.setVariables(variables);
+        this.jobInfo.setVariables(this.getVariablesAsReplacementMap());
+    }
```

Comment:
why not. I think you should do the same with genericInformation. I tested that the jobInfo contains a null generic info when it should not.

Reply:
addressed @fviale